### ISSUE TITLE
[CI] SearchableSnapshotActionIT testSearchableSnapshotsInHotPhasePinnedToHotNodes failing

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#86890]
**Build scan:**
https://gradle-enterprise.elastic.co/s/pzurvgcutg7gc/tests/:x-pack:plugin:ilm:qa:multi-node:javaRestTest/org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT/testSearchableSnapshotsInHotPhasePinnedToHotNodes

**Reproduction line:**
`gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testSearchableSnapshotsInHotPhasePinnedToHotNodes" -Dtests.seed=6E72377C9BDFDFDB -Dtests.locale=de-LU -Dtests.timezone=Etc/GMT-14 -Druntime.java=18`

**Applicable branches:**
8.2

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT&tests.test=testSearchableSnapshotsInHotPhasePinnedToHotNodes

**Failure excerpt:**
```
org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:65092], URI [_snapshot/HZBbwxlidMtpddHMCFQ], status line [HTTP/1.1 409 Conflict]
{"error":{"root_cause":[{"type":"repository_conflict_exception","reason":"[HZBbwxlidMtpddHMCFQ] trying to modify or unregister repository that is currently used (found 1 searchable snapshots indices that use the repository: [restored-.ds-logs-dhemcylega-2022.05.18-000002/_Gldrg5jSPmxdvZzcFPU2w])"}],"type":"repository_conflict_exception","reason":"[HZBbwxlidMtpddHMCFQ] trying to modify or unregister repository that is currently used (found 1 searchable snapshots indices that use the repository: [restored-.ds-logs-dhemcylega-2022.05.18-000002/_Gldrg5jSPmxdvZzcFPU2w])"},"status":409}

  at __randomizedtesting.SeedInfo.seed([6E72377C9BDFDFDB:1B98CD4E21AE466A]:0)
  at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:346)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:312)
  at org.elasticsearch.client.RestClient.performRequest(RestClient.java:287)
  at org.elasticsearch.test.rest.ESRestTestCase.deleteRepository(ESRestTestCase.java:1042)
  at org.elasticsearch.test.rest.ESRestTestCase.wipeSnapshots(ESRestTestCase.java:1034)
  at org.elasticsearch.test.rest.ESRestTestCase.lambda$wipeCluster$3(ESRestTestCase.java:666)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1053)
  at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:665)
  at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:399)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

**Comment by fcofdez** [comment]
Another instance of this https://gradle-enterprise.elastic.co/s/jra2oazwy3ip4/tests/:x-pack:plugin:ilm:qa:multi-node:javaRestTest/org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT/testSearchableSnapshotsInHotPhasePinnedToHotNodes

**Comment by kingherc** [comment]
Another test failure [here](https://gradle-enterprise.elastic.co/s/c4xstlmkfrbic/console-log?task=:x-pack:plugin:ilm:qa:multi-node:javaRestTest) for 7.17. Probably PR needs to be backported. Re-opening.

```
:x-pack:plugin:ilm:qa:multi-node:javaRestTest FAILED |  
  | Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Dsun.jnu.encoding=UTF8 |  
  |   |  
  |   |  
  | org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT > testSearchableSnapshotsInHotPhasePinnedToHotNodes FAILED |  
  | org.elasticsearch.client.ResponseException: method [DELETE], host [http://127.0.0.1:62970], URI [_snapshot/ADFesFCMggtZd], status line [HTTP/1.1 500 Internal Server Error] |  
  | {"error":{"root_cause":[{"type":"illegal_state_exception","reason":"trying to modify or unregister repository [ADFesFCMggtZd] that is currently used (found 1 searchable snapshots indices that use the repository: [restored-.ds-logs-qiiqtcjuml-2022.11.07-000002/E-109GiwSxeiFTRhYQfAIA])"}],"type":"illegal_state_exception","reason":"trying to modify or unregister repository [ADFesFCMggtZd] that is currently used (found 1 searchable snapshots indices that use the repository: [restored-.ds-logs-qiiqtcjuml-2022.11.07-000002/E-109GiwSxeiFTRhYQfAIA])"},"status":500} |  
  | at __randomizedtesting.SeedInfo.seed([E6FE75C26709D13:7B851D6E9C0104A2]:0) |  
  | at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:347) |  
  | at org.elasticsearch.client.RestClient.performRequest(RestClient.java:313) |  
  | at org.elasticsearch.client.RestClient.performRequest(RestClient.java:288) |  
  | at org.elasticsearch.test.rest.ESRestTestCase.deleteRepository(ESRestTestCase.java:1030) |  
  | at org.elasticsearch.test.rest.ESRestTestCase.wipeSnapshots(ESRestTestCase.java:1022) |  
  | at org.elasticsearch.test.rest.ESRestTestCase.lambda$wipeCluster$4(ESRestTestCase.java:648) |  
  | at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1129) |  
  | at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:647) |  
  | at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:384) |  
  | at sun.reflect.GeneratedMethodAccessor16.invoke(Unknown Source) |  
  | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) |  
  | at java.lang.reflect.Method.invoke(Method.java:498) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) |  
  | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  | at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) |  
  | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891) |  
  | at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902) |  
  | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41) |  
  | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  | at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53) |  
  | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47) |  
  | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  | at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54) |  
  | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390) |  
  | at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850) |  
  | at java.lang.Thread.run(Thread.java:748) |  
  | REPRODUCE WITH: gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testSearchableSnapshotsInHotPhasePinnedToHotNodes" -Dtests.seed=E6FE75C26709D13 -Dtests.locale=th -Dtests.timezone=Asia/Shanghai -Druntime.java=8 |  
 

<br class="Apple-interchange-newline">:x-pack:plugin:ilm:qa:multi-node:javaRestTest FAILED	
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8 -Dsun.jnu.encoding=UTF8	
org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT > testSearchableSnapshotsInHotPhasePinnedToHotNodes FAILED	
    org.elasticsearch.client.ResponseException: method [DELETE], host [http://127.0.0.1:62970/], URI [_snapshot/ADFesFCMggtZd], status line [HTTP/1.1 500 Internal Server Error]	
    {"error":{"root_cause":[{"type":"illegal_state_exception","reason":"trying to modify or unregister repository [ADFesFCMggtZd] that is currently used (found 1 searchable snapshots indices that use the repository: [restored-.ds-logs-qiiqtcjuml-2022.11.07-000002/E-109GiwSxeiFTRhYQfAIA])"}],"type":"illegal_state_exception","reason":"trying to modify or unregister repository [ADFesFCMggtZd] that is currently used (found 1 searchable snapshots indices that use the repository: [restored-.ds-logs-qiiqtcjuml-2022.11.07-000002/E-109GiwSxeiFTRhYQfAIA])"},"status":500}	
        at __randomizedtesting.SeedInfo.seed([E6FE75C26709D13:7B851D6E9C0104A2]:0)	
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:347)	
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:313)	
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:288)	
        at org.elasticsearch.test.rest.ESRestTestCase.deleteRepository(ESRestTestCase.java:1030)	
        at org.elasticsearch.test.rest.ESRestTestCase.wipeSnapshots(ESRestTestCase.java:1022)	
        at org.elasticsearch.test.rest.ESRestTestCase.lambda$wipeCluster$4(ESRestTestCase.java:648)	
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1129)	
        at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:647)	
        at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:384)	
        at sun.reflect.GeneratedMethodAccessor16.invoke(Unknown Source)	
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	
        at java.lang.reflect.Method.invoke(Method.java:498)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:1004)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)	
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)	
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)	
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)	
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)	
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)	
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)	
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)	
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)	
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)	
        at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)	
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)	
        at java.lang.Thread.run(Thread.java:748)	
REPRODUCE WITH: gradlew ':x-pack:plugin:ilm:qa:multi-node:javaRestTest' --tests "org.elasticsearch.xpack.ilm.actions.SearchableSnapshotActionIT.testSearchableSnapshotsInHotPhasePinnedToHotNodes" -Dtests.seed=E6FE75C26709D13 -Dtests.locale=th -Dtests.timezone=Asia/Shanghai -Druntime.java=8
```

**Comment by arteam** [comment]
Closed via #91805
